python/src/dataset/io_stats.rs

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+//! IO statistics tracking for dataset operations
+
+use pyo3::{pyclass, pymethods};
+
+/// IO statistics for dataset operations
+///
+/// This tracks the number of IO operations and bytes transferred for read and write
+/// operations performed on the dataset's object store.
+///
+/// Note: Calling `io_stats()` returns the statistics accumulated since the last call
+/// and resets the internal counters (incremental stats pattern).
+#[pyclass(name = "IOStats", module = "_lib", get_all)]
+#[derive(Clone, Debug)]
+pub struct IoStats {
```

Comment:
It would be nice if we could include these docs in mkdocs.  I'll make an issue to figure that out.  Then we wouldn't need the lengthy `Returns` block on the python.